docs/customresourcestate-metrics.md

```diff
@@ -272,19 +267,55 @@ spec:
-              path: [metadata, annotations]
-# This will output the following metric:
-# HELP kube_customresource_autoscaling_annotations Kubernetes annotations converted to Prometheus labels.
-# TYPE kube_customresource_autoscaling_annotations gauge
-# kube_customresource_autoscaling_annotations{customresource_group="autoscaling.k8s.io", customresource_kind="VerticalPodAutoscaler", customresource_version="v1", namespace="default",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",verticalpodautoscaler="hamster-vpa"} 123
+              path: [status, recommendation, containerRecommendations]
+              valueFrom: [target, memory]
+              labelsFromPath:
+                container: [containerName]
+          commonLabels:
+            resource: "memory"
+            unit: "byte"
+        - name: "vpa_containerrecommendations_target"
+          help: "VPA container recommendations for cpu."
+          each:
+            type: Gauge
+            gauge:
+              path: [status, recommendation, containerRecommendations]
+              valueFrom: [target, cpu]
+              labelsFromPath:
+                container: [containerName]
+          commonLabels:
+            resource: "cpu"
+            unit: "core"
+```
+
+The above configuration was tested on [this](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/examples/hamster.yaml) VPA configuration, outputting the following metrics:
+
+```
+# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for memory.
+kube_customresource_vpa_containerrecommendations_target{container="hamster",customresource_group="autoscaling.k8s.io",customresource_kind="VerticalPodAutoscaler",customresource_version="v1",namespace="default",resource="memory",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",unit="byte",verticalpodautoscaler="hamster-vpa"} 1.048576e+08
+
+# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for cpu.
+kube_customresource_vpa_containerrecommendations_target{container="hamster",customresource_group="autoscaling.k8s.io",customresource_kind="VerticalPodAutoscaler",customresource_version="v1",namespace="default",resource="cpu",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",unit="core",verticalpodautoscaler="hamster-vpa"} 0.1
 ```
 
-The above configuration was tested on [this](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/examples/hamster.yaml) VPA configuration, with an added annotation (`foo: 123`).
+In RBAC-enabled clusters and to ensure that the VPA Metrics will be captured seamlessly, make sure that the Cluster Role / Role assigned to the Kube State Metrics has permissions to `list` & `watch` VPA resources (as well as `customresourcedefinitions`) e.g.:
+
+```yaml
+- verbs:
+    - list
+    - watch
+  apiGroups:
```

Comment:
Usually permissions are set per apigroup/resource Otherwise you would give permission to an (unlikely, but possible) customresourcedefinitions.autoscaling.k8s.io or verticalpodautoscalers.apiextensions.k8s.io which you did not intend to provide.